### PR TITLE
Allow configuring docker machine egress rules

### PR DESCRIPTION
**Pull request by mhulscher**
## Description

I would like to be able to configure the egress rules of the docker-machine security group. This PR introduces a new variable called `docker_machine_egress_rules` to configure these egress rules. It defaults to allow all egress traffic to maintain backwards compatibility.

## Migrations required

YES, for a smooth upgrade the security group for docker machine needs to be tainted

```bash
terraform taint module.runner.aws_security_group.docker_machine
```

## Verification

Ran a `terraform plan` in the `runner-docker` example.

## Documentation

Ran pre-commit with latest version of each tool; this also re-generates larges portions of the README.

### CODE REVIEW

**Comment by npalm** [comment]
Please can you check the failing CI job, the modules is still on version 0.13.6. Assume you have used another terraform version. We use tfenv ot set the used terraform version.  Examples are verified with multiple terraform versions

**Comment by mhulscher** [comment]
Hi @npalm ,

It looks like `terraform fmt`'s opinion about what is correct formatting has changed since terraform 0.13.6. When I run `terraform fmt` against my current PR, terraform recommends that I remove the whitespace between the character `!` and the next term in a negation. However, newer terraform versions recommend the exact opposite.

```sh
❯ ~/.bins/terraform-0.13.6 fmt -recursive -check=true -diff             
main.tf
--- old/main.tf
+++ new/main.tf
@@ -38,7 +38,7 @@
 }
 
 locals {
-  enable_asg_recreation = var.enable_forced_updates != null ? !var.enable_forced_updates : var.enable_asg_recreation
+  enable_asg_recreation = var.enable_forced_updates != null ? ! var.enable_forced_updates : var.enable_asg_recreation
 
   template_user_data = templatefile("${path.module}/template/user-data.tpl",
     {
@@ -125,7 +125,7 @@
       runners_root_size                 = var.runners_root_size
       runners_iam_instance_profile_name = var.runners_iam_instance_profile_name
       runners_use_private_address_only  = var.runners_use_private_address
-      runners_use_private_address       = !var.runners_use_private_address
+      runners_use_private_address       = ! var.runners_use_private_address
       runners_request_spot_instance     = var.runners_request_spot_instance
       runners_environment_vars          = jsonencode(var.runners_environment_vars)
       runners_pre_build_script          = var.runners_pre_build_script
```

If I update `main.tf` to please 0.13.6, newer versions will start complaining. Do you have any suggestions what to do with this?

**Comment by npalm** [comment]
Please update to please the current TF version for the module, I will upgrade sources and formating asap to a newer version

**Comment by mhulscher** [comment]
@npalm done!

File: variables.tf [hunk 1]
```diff
@@ -705,3 +705,29 @@ variable "docker_machine_iam_policy_arns" {
...
+  default = [{
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+    prefix_list_ids  = null
+    from_port        = 0
+    protocol         = "-1"
+    security_groups  = null
+    self             = null
+    to_port          = 0
+    description      = null
```
Comment (npalm) [inline variables.tf:731]:
```suggestion
    description      = "Allow all egress traffic for docker machine build runners"
```

**Review by npalm — CHANGES_REQUESTED** [review]
Made a small suggestion for a change. Furthermore there is a small migration required. I ran a upgrade on the default example, which fails. After tainting the security group an upgrade was smooth.

```bash
terraform taint module.runner.aws_security_group.docker_machine
```

Please can you update the PR description with a small not for the required migration

**Review by npalm — APPROVED** [review]
Updated description, and made a smaal change in the default description of the egress rule.

**Pull request merged**